src/Reader.php

```diff
@@ -52,6 +52,20 @@ public function query(callable $callable = null)
         $iterator = $this->applyIteratorFilter($iterator);
         $iterator = $this->applyIteratorSortBy($iterator);
         $iterator = $this->applyIteratorInterval($iterator);
+
```

Comment:
Ah yes, we want to respond to the original set, not the filtered one. Will implement.